omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1991,3 +1997,24 @@ def _settle_robot(self):
                 break
             empty_action = self._empty_action()
             yield self._postprocess_action(empty_action)
+
+    def linspace_1d_tensor(self, start_pos, target_pose, num_poses):
+        """
+        Create evenly spaced samples between two 1D tensors.
+
+        :param start_pos: Starting 1D tensor
+        :param target_pose: Ending 1D tensor
+        :param num_poses: Number of poses (samples) to generate
+        :return: Tensor of shape (num_poses, dim) where dim is the dimension of input tensors
+        """
+        # Ensure inputs are 1D tensors
+        assert start_pos.dim() == 1 and target_pose.dim() == 1, "Input tensors must be 1D"
+        assert start_pos.shape == target_pose.shape, "Input tensors must have the same shape"
+
+        # Create a tensor of interpolation factors
+        t = th.linspace(0, 1, num_poses)
+
+        # Perform the interpolation
+        interpolated_points = start_pos.unsqueeze(0) + (target_pose - start_pos).unsqueeze(0) * t.unsqueeze(1)
+
+        return interpolated_points
```

Comment:
Move to a util file somewhere. Actually also ask @hang-yin if he already did this bc I doubt this is the only use of linspace.